packages/bull-vault/test/e2e/BullStrategyTestFork.t.sol

```diff
@@ -1,43 +1,179 @@
+        IERC20(crabV2).transfer(user1, 100e18);
+        // some WETH and USDC rich address
+        vm.prank(0x57757E3D981446D585Af0D9Ae4d7DF6D64647806);
+        IERC20(weth).transfer(user1, 10000e18);
+    }
+
+    function testInitialDeposit() public {
+        uint256 crabToDeposit = 10e18;
+        (uint256 wethToLend, uint256 usdcToBorrow) = _calcCollateralAndBorrowAmount(crabToDeposit);
+
+        vm.startPrank(user1);
+        IERC20(crabV2).approve(address(bullStrategy), crabToDeposit);
+        bullStrategy.deposit{value: wethToLend}(crabToDeposit);
+        vm.stopPrank();
+
+        assertEq(bullStrategy.balanceOf(user1), crabToDeposit);
+        assertEq(IEulerDToken(dToken).balanceOf(address(bullStrategy)), usdcToBorrow);
+        assertTrue(wethToLend.sub(IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy))) <= 1);
+        assertEq(IERC20(usdc).balanceOf(user1), usdcToBorrow);
+    }
+
+    // fuzz testing is awesome!
+    function testFuzzingDeposit(uint256 _crabAmount) public {
+        vm.assume(_crabAmount < IERC20(crabV2).balanceOf(user1));
+        vm.assume(_crabAmount > 0);
+
+        uint256 bullToMint = _calcBullToMint(_crabAmount);
+        (uint256 wethToLend, uint256 usdcToBorrow) = _calcCollateralAndBorrowAmount(_crabAmount);
+        uint256 userBullBalanceBefore = bullStrategy.balanceOf(user1);
+        uint256 ethInLendingBefore = IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy));
+        uint256 usdcBorrowedBefore = IEulerDToken(dToken).balanceOf(address(bullStrategy));
+        uint256 userUsdcBalanceBefore = IERC20(usdc).balanceOf(user1);
+
+        vm.startPrank(user1);
+        IERC20(crabV2).approve(address(bullStrategy), _crabAmount);
+        bullStrategy.deposit{value: wethToLend}(_crabAmount);
+        vm.stopPrank();
+        
+        assertEq(bullStrategy.balanceOf(user1).sub(userBullBalanceBefore), bullToMint);
+        assertTrue(wethToLend.sub(IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy)).sub(ethInLendingBefore)) <= 2);
+        assertEq(IEulerDToken(dToken).balanceOf(address(bullStrategy)).sub(usdcBorrowedBefore), usdcToBorrow);
+        assertEq(IERC20(usdc).balanceOf(user1).sub(userUsdcBalanceBefore), usdcToBorrow);
+    }
+
+    function _getCrabVaultDetails() internal view returns (uint256, uint256) {
+        VaultLib.Vault memory strategyVault = IController(address(controller)).vaults(crabV2.vaultId());
+
+        return (strategyVault.collateralAmount, strategyVault.shortAmount);
+    }
+
+    function _calcBullToMint(uint256 _crabToDeposit) internal view returns (uint256) {
+        if (IERC20(bullStrategy).totalSupply() == 0) {
+            return _crabToDeposit;
+        } else {
+            uint256 share = _crabToDeposit.wdiv(IERC20(crabV2).balanceOf(address(bullStrategy)));
+            uint256 bullTotalSupply = bullStrategy.totalSupply();
+            return share.wmul(bullTotalSupply).wdiv(uint256(1e18).sub(bullTotalSupply));
```

Comment:
needs to be updated to take into account the fix to make the denominator be (1-share) not 1-bullTotalSupply

Reply:
fixed